### PR TITLE
feat(wallet): Add new saved addresses

### LINKED ISSUE #19764: #️⃣ Save a new address in wallet settings
# Feature Issue

### User Story

#️⃣ As a user, 
I want to save a hexadecimal ethereum address to the list of my saved addresses

### Description

Implement the flow of saving an address. ENS names are not to be cared for in this issue.

### Acceptance Criteria

| Screen | Description | Acceptance Criteria |
|--------|--------|--------|
| image | This screen is not a part of this issue. It only shows the entry point to the flow | NA |
| image | When the user clicks plus button on the saved addresses screen, this screen to add an address should open | 1. The scan button is not to be implemented in this issue<br/>2. The CTA should be disabled <br/>3. Adding an address or an ens to the input box should verify the shape of the address. If the address is valid, the CTA should be enabled<br/>4. If the address is already saved, or invalid or belongs to user, there should be an error (shown in the screens below)<br>5. The input should be auto-focused and the keyboard should open automatically when the user enters the screen<br/>6. Adding an ENS is **not** a part of this issue |
| image | My address error case for saving own address | 1. A user should not be able to save her own address (any address for which the user has accounts for) |
| image |   | 1. If the address in not a valid ethereum address, an error should be shown | 
| image | | 1. If the address is already saved, we show the address. Clicking on the Saved address should take you to the Saved Address Detail screen |
| image | Valid address | 1. When the address is valid, the CTA should be enabled<br/>2. The network names do **not** need to be color coded in this iteration |
| image | If address is correct, we open the screen to save the address | 1. User should be add label for the address <br/>2. User should be able to set a color for address<br/>3. User should be able to select a network for the address<br/>4. [Subsequent screens in this flow](https://www.figma.com/file/hLFPHYnuM0pOzgiabtqQ6k/Descoped-Wallet-settings-for-Mobile?type=design&node-id=80-169810&mode=design&t=DJXQe9W3moVlqy0c-4) | 
| image | Success toast | 1. On saving address successfully, the user should be redirected to Saved Addresses list<br/>2. A success toast should be toasted |

### Notes

Figma Flow: https://www.figma.com/file/hLFPHYnuM0pOzgiabtqQ6k/Descoped-Wallet-settings-for-Mobile?type=design&node-id=80-168215&mode=design&t=DJXQe9W3moVlqy0c-4

 In the designs, the user is not allowed to save an address if the address being saved is a contact. But this condition is not needed. I confirmed with Alisher before writing this issue.

### PR DESCRIPTION
**Pull request by smohamedjavid**
fixes #19764

### Summary

This PR: 
- Add flow for adding new saved addresses from settings
- Adds validations for entered address

https://github.com/status-im/status-mobile/assets/19339952/29502fab-138b-4a7c-9fab-983e7e11338f

### Review notes

- This PR covers only adding with an address
- Adding with ENS or scanning an address will be done separately
- Selecting network preferences will be done separately

### Testing notes

Testing will be done once the saved addresses epic is completed

#### Platforms

- Android
- iOS

status: ready

### CODE REVIEW

File: src/status_im/contexts/settings/wallet/saved_addresses/events_test.cljs [hunk 1]
```diff
@@ -150,3 +150,21 @@
...
+
+(deftest add-saved-address-success-test
+  (testing "add saved address success test - gets saved addresses, dismiss modals and dispatch toast"
+    (let [cofx          {:db {}}
+          toast-message "Address saved"
+          effects       (events/add-saved-address-success cofx [toast-message])
+          result-fx     (:fx effects)
+          expected-fx   [[:dispatch [:wallet/get-saved-addresses]]
+                         [:dispatch [:dismiss-modal :screen/settings.add-address-to-save]]
+                         [:dispatch [:dismiss-modal :screen/settings.save-address]]
```
Comment (seanstrom) [inline src/status_im/contexts/settings/wallet/saved_addresses/events_test.cljs]: Nice, I gotta start using this for closing my modals 👍

**Review by seanstrom — APPROVED** [review]

**Review by mohsen-ghafouri — APPROVED** [review]
Looks perfect.

I have only some suggestion about folder structure and also about moving validations to separate file but we can discuss them first and then do it in another PR.

**Comment by smohamedjavid** [comment]
@status-im/mobile-qa - Can you help me quickly check the failed test cases? 🙏

**Comment by VolodLytvynenko** [comment]
Hi @smohamedjavid!
The e2e results are reviewed, the failed tests are not related to the PR.
PR can proceed to the next stage.

**Pull request merged**